tests/unit-tests/test-class-usage-tracking.php

```diff
@@ -0,0 +1,318 @@
+		} catch ( WP_Die_Exception $e ) {
+			$wp_die_args = $e->get_wp_die_args();
+			$this->assertEquals( array(), $wp_die_args['args'] );
+		}
+
+		// Refresh settings
+		Sensei()->settings->get_settings();
+
+		$this->assertTrue( Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertTrue( get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+	}
+
+	/**
+	 * Ensure tracking is disabled through ajax request.
+	 */
+	public function test_ajax_request_disable_tracking() {
+		$this->setup_ajax_request();
+		$_POST['enable_tracking'] = '0';
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertFalse( !! get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+
+		try {
+			$this->usage_tracking->handle_tracking_opt_in();
+		} catch ( WP_Die_Exception $e ) {
+			$wp_die_args = $e->get_wp_die_args();
+			$this->assertEquals( array(), $wp_die_args['args'] );
+		}
+
+		// Refresh settings
+		Sensei()->settings->get_settings();
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertTrue( get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+	}
+
+	/**
+	 * Ensure ajax request fails on nonce failure and does not update option.
+	 */
+	public function test_ajax_request_failed_nonce() {
+		$this->setup_ajax_request();
+		$_REQUEST['nonce'] = 'invalid_nonce_1234';
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertFalse( !! get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+
+		try {
+			$this->usage_tracking->handle_tracking_opt_in();
+		} catch ( WP_Die_Exception $e ) {
+			$wp_die_args = $e->get_wp_die_args();
+			$this->assertEquals( 403, $wp_die_args['args']['response'] );
+		}
+
+		// Refresh settings
+		Sensei()->settings->get_settings();
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertFalse( !! get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+	}
+
+	/**
+	 * Ensure ajax request fails on authorization failure and does not update option.
+	 */
+	public function test_ajax_request_failed_auth() {
+		$this->setup_ajax_request();
+
+		$user = wp_get_current_user();
+		$user->remove_cap( 'manage_sensei' );
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertFalse( !! get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+
+		try {
+			$this->usage_tracking->handle_tracking_opt_in();
+		} catch ( WP_Die_Exception $e ) {
+			$wp_die_args = $e->get_wp_die_args();
+			$this->assertEquals( 403, $wp_die_args['args']['response'] );
+		}
+
+		// Refresh settings
+		Sensei()->settings->get_settings();
+
+		$this->assertFalse( !! Sensei()->settings->get( 'sensei_usage_tracking_enabled' ) );
+		$this->assertFalse( !! get_option( 'sensei_usage_tracking_opt_in_hide' ) );
+	}
+
+	/**
+	 * Helpers for ajax request.
+	 */
+	private function setup_ajax_request() {
```

Comment:
I'm not sure I understand what part of the usage tracking code this is testing since it doesn't appear to be calling any usage tracking functions?

Reply:
It's a helper method, not a test method. I moved it to the bottom of the file to make it more clear (as opposed to being at the end of the section where it is used)

Reply:
Ah, right. Thx.